### PR TITLE
Hibernate reactive panache

### PR DESCRIPTION
**Pull request by DavideD**
A quickstart using Panache with Hibernate Reactive.
Note that I'm flushing the CRUD operations manually but I'm wondering if there is a way to support the `@Transactional` annotation.

Anyway, it seems to work and it should help people wanting to play with the experimental extension.

By the way, all the quickstarts are using Postgres 10.5, shouldn't we upgrade to 13?

Cheers

### CODE REVIEW

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/Fruit.java
Comment (FroMage) [inline hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/Fruit.java:14]: This should be a public field with no accessors: they're generated.

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (FroMage) [inline hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:78]: We have `deleteById`, would that be better?

File: README.md
Comment (FroMage) [inline README.md:47]:
```suggestion
* [Hibernate Reactive Panache and RESTEasy Reactive](./hibernate-reactive-quickstart): Exposing a CRUD service over REST using Hibernate Reactive and Panache to connect to a PostgreSQL database
```

File: hibernate-reactive-panache-quickstart/README.md
Comment (FroMage) [inline hibernate-reactive-panache-quickstart/README.md:1]:
```suggestion
# Quarkus demo: Hibernate Reactive with Panache and RESTEasy Reactive
```

File: hibernate-reactive-panache-quickstart/README.md
Comment (FroMage) [inline hibernate-reactive-panache-quickstart/README.md:7]:
```suggestion
 - RESTEasy Reactive to expose the REST endpoints
```

**Review by FroMage — CHANGES_REQUESTED** [review]
LGTM except minor comments

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:78]: Is there a way to flush without having the entity?

Comment (FroMage) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:78]: https://github.com/quarkusio/quarkus/issues/15595

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:78]: What I need here is something like `persistAndFlush`. It seems there is no  equivalent for the delete.
I could change the code to something like this:
```
    @DELETE
    @Path("{id}")
    public Uni<Response> delete(@RestPath Long id) {
        return Fruit.deleteAndFlushById(id)
                .onItem().transform( deleted -> deleted
                        ? Response.ok().status(204).build()
                        : Response.ok().status(404).build()
        );
    }
```

Comment (FroMage) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:78]: Good point. I'll add it to the issue.

**Review by FroMage — APPROVED** [review]
Great, thanks!

**Comment by FroMage** [comment]
> What I'm wondering now if it would be better to wrap everything in a transaction using Panache.withTransaction instead of using flush. What do you think?

Hey, you're the DB expert, I defer to you ;)

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java]: > Hey, you're the DB expert, I defer to you ;)

:) Yeah, I think we need to show how to create transactions

**Comment by DavideD** [comment]
OK, I've updated the example and now it uses transactions.
I think it looks pretty good now.

Thanks

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (gavinking) [inline hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:61]: @DavideD `onItem().transform()` is a rather verbose way to do this. I think it's better to use the new methods we added to Mutiny.

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (gavinking) [inline hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: Should use `invoke()` here.

**Review by gavinking — CHANGES_REQUESTED** [review]
Looks to me like this code example is using all the old Mutiny APIs, not the new stuff we added to make this sort of code look prettier.

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: I need the entity so that I can add it to the response

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: I suppose I could return the `Response` directly

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: I will change it to this:
```
       return Panache
                .withTransaction(() -> Fruit.<Fruit> findById(id)
                        .onItem().ifNotNull().transform(entity -> {
                            entity.name = fruit.name;
                            return Response.ok(entity).build();
                        }))
                .onItem().ifNull().continueWith(() -> Response.ok().status(NOT_FOUND).build());
```

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: But I think I prefer the other approach as it keeps the update of the entity and the return of the response separated

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:61]: Ah ok, found it... `replaceWith`

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:61]: How come they have such a different name when they do the same thing?

Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:61]: Ah ok, replace doesn't have the argument.

**Comment by DavideD** [comment]
I've added a commit with the changes: https://github.com/quarkusio/quarkus-quickstarts/pull/809/commits/7d56eadc6201ac8443fa2949e1d89526b0b8558a

File: hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java
Comment (DavideD) [reply hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java:75]: If I use `invoke` on the first line then I loose the entity.
Maybe I simply don't see it.

> We need to get away from using transform(), IMO.

But why? This to me seems a reasonable use case to me

**Comment by DavideD** [comment]
I've updated the PR but there is still a `transform` that I don't think we can remove.
I'm happy to be proven wrong.

**Comment by gavinking** [comment]
LGTM, thanks @DavideD!

**Comment by FroMage** [comment]
@gavinking you need to approve the changes, if LGTM means that ;)

**Comment by FroMage** [comment]
Well I can still merge it, though…

**Review by gavinking — APPROVED** [review]

**Pull request merged**
